src/Cli/dotnet/CommonOptions.cs

```diff
@@ -263,6 +262,18 @@ private static IEnumerable<string> ForwardSelfContainedOptions(bool isSelfContai
             parseResult.HasOption(ArchitectureOption) ||
             parseResult.HasOption(LongFormArchitectureOption) ||
             parseResult.HasOption(OperatingSystemOption);
+
+        internal static Option<T> AddCompletions<T>(this Option<T> option, Func<CompletionContext, IEnumerable<CompletionItem>> completionSource)
+        {
+            option.CompletionSources.Add(completionSource);
+            return option;
```

Comment:
this extension method got removed from S.CL, it now returns `void`. Based on the usages I decided that implement it here so it can be used in the SDK the way it was so far (fewer changes)